### PR TITLE
add WebHomeMiniAvatarUploadButton Interaction

### PR DESCRIPTION
**Pull request by justjanne**
Related: https://github.com/vector-im/element-web/issues/21946
Dependents: https://github.com/matrix-org/matrix-react-sdk/pull/8474

## In Short
* Adds new WebHomeMiniAvatarUploadButton interaction

### CODE REVIEW

**Review by t3chguy — APPROVED** [review]

**Comment by justjanne** [comment]
As "interaction" doesn’t really fit for an avatar upload, the other option that I can see right now would be adding a new event type, e.g.:

```json
{
  "type": "object",
  "description": "Triggered when a user uploads an avatar.",
  "properties": {
    "eventName": {
      "enum": ["AvatarUpload"]
    },
    "target": {
      "description": "Whose avatar is being changed.",
      "type": "string",
      "oneOf": [
        { "const": "Room", "description": "User replaced the avatar of a room." },
        { "const": "User", "description":  "User replaced their own avatar." }
      ]
    },
    "isInitial": {
      "description": "Whether the target was without avatar before this upload.",
      "type": "boolean"
    },
    "source": {
      "description": "UI element which caused the upload",
      "type": "string",
      "oneOf": [
        { "const": "HomePage", "description": "New user welcome screen" },
        { "const": "NewRoomIntro", "description": "New room welcome message" },
        { "const": "RoomProfileSettings", "description": "Room settings dialog" },
        { "const": "SpaceBasicSettings", "description": "Space settings dialog" },
        { "const": "ProfileSettings", "description": "Profile settings dialog" }
      ]
    }
  },
  "required": ["eventName", "target", "isInitial", "source"],
  "additionalProperties": false
}
```

Is that preferred? If so, I’ll update both PRs accordingly

**Comment by t3chguy** [comment]
I think we're wanting to avoid very pidgeonholed events so it doesn't become overwhelming

**Comment by justjanne** [comment]
We’re now tracking the onClick event of the upload button, as previously discussed :)

File: schemas/Interaction.json [hunk 1]
```diff
@@ -44,6 +44,7 @@
         {"const": "WebRoomListHeaderPlusMenuCreateChatItem", "description": "User clicked the create DM button in the + context menu of the room list header in Element Web/Desktop." },
         {"const": "WebSpaceContextMenuHomeItem", "description": "User clicked the home button in the context menu of a space in Element Web/Desktop." },
         {"const": "WebHomeExploreRoomsButton", "description": "User clicked the explore rooms button in the home page of Element Web/Desktop." },
+        {"const": "WebHomeMiniAvatarUploadButton", "description": "User updated their avatar via the mini avatar uploader in the home page of Element Web/Desktop." },
```
Comment (t3chguy) [inline schemas/Interaction.json:47]: this description is no longer right

Comment (justjanne) [reply schemas/Interaction.json:47]: Thanks for catching that, I had totally missed it!

**Review by t3chguy — APPROVED** [review]

**Pull request merged**